datafusion/core/src/execution/context/mod.rs

```diff
@@ -1790,6 +1837,38 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_dynamic_file_query() -> Result<()> {
+        let path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let path = path.join("tests/tpch-csv/customer.csv");
+        let url = format!("file://{}", path.display());
+        let cfg = SessionConfig::new().set_str("datafusion.catalog.has_header", "true");
```

Comment:
likewise here it isn't necessary to set the `has_header` flag anymore I don't think